### DISCUSSION TITLE
/hyndsight/flat-forecasts/

### DISCUSSION THREAD

**Discussion by robjhyndman** [robjhyndman/robjhyndman.com#120] [Announcements]
Flat forecasts 
 https://robjhyndman.com/hyndsight/flat-forecasts/

**Comment by robjhyndman** [comment]
```header
Originally posted by buggyfunbunny on 2012-08-20 18:28:39
````
There\'s an olde wive\'s tale (or may be old crony\'s) about the
consultant and the client, to wit\...

client: Mr. Consultant, please tell me the time! I don\'t know what time
it is!\
consultant: Client, give me your hand, there there, all will be well.
\[client proffers left hand, consultant pushes back sleeve, revealing
watch.\] Client, it is 10:07 AM\
client: Ah!!! Thank you so much. I was so worried I knew not the time.

**Comment by robjhyndman** [comment]
```header
Originally posted by ericstrom on 2012-08-21 03:10:35
````
Rob, would a compromise be possible ? Forecast function provides a note
as to why the flat forecast (e.g. note the flat forecast due to random
walk model being most appropriate).

By the way, great book and great packages. Thanks much !

**Comment by robjhyndman** [comment]
```header
Originally posted by dan_ on 2012-08-29 13:28:42
````
Rob, this is an excellent statement. In addition it is so clear and so
to the point, simply. Thank you for posting it.

**Comment by robjhyndman** [comment]
```header
Originally posted by dp_planner on 2013-02-25 13:02:03
````
I have just come across this article. I provide forecasts to clients for
inventory optimization. When it comes to client, you have to sell the
forecast. The two main points of sell is credibility and accuracy. For a
long term out look (more than 2-3 months out), a flat forecast is not
credible since it is there has to be some type of variation in the
business going forward. Having said that, a flat forecast for a current
month forecast, is the most credible (from the clients perspective).
That\'s the constant battle with the client.

**Comment by robjhyndman** [comment]
```header
Originally posted by Vasily Nekrasov on 2017-07-19 09:06:54
````
I still think that flat a forecast by arima is, in a sense, bug.

Consider the following test-code:\
If you left &quot;0.8\*x\[i-24\]&quot; commented, you get an Arima without a
seasonal component and the forecast is very flat.\
If you uncomment it, there is 24-periodicity and the forecast seems much
more plausible (it flattens quick but not immediately).

That\'s why in practice I sometimes use STL even for non-seasonal and
non-trendy models.\
The forecasts made by STL are theoretically invalid but in practice are
better than from ARIMA.

x = rnorm(1008, 0,1)\
y = array(0.0, dim=1008)\
y\[1\] = x\[1\]\
for(i in 25:1008)\
{\
y\[i\] = 0.6\*y\[i-1\] + x\[i\] #uncomment it to get periodicity+
0.8\*x\[i-24\]\
}\
par(mfrow=c(1,2))

auto.arima(y) #retuns unfeasable results\
armaMod = Arima(y, order=c(1,0,0), seasonal=list(order=c(1,0,0),
period=24)) #much better\
prog = forecast(armaMod, 100)\
plot(prog)

stlMod = stl(ts(y, frequency=24), s.windo=&quot;period&quot;)\
#plot(stlMod)\
prog = forecast(stlMod, 100, method=&quot;arima&quot;)\
plot(prog)

[https://uploads.disquscdn.c\...](https://uploads.disquscdn.com/images/cf10f5adfa271b612b7553f52028970042472d697609dea010e8dd2fcb1651ed.png)
[https://uploads.disquscdn.c\...](https://uploads.disquscdn.com/images/0c387289de3b42de72a3c5ae8ab3985122fa43ef97220b3535de5acc0db60b27.png)

**Reply by robjhyndman** [reply]
```header
Originally posted on 2017-07-19 09:28:32
````
A forecast is a mean, not a sample path. The mean can be flat even if
the sample paths are not.

**Reply by robjhyndman** [reply]
```header
Originally posted by Vasily Nekrasov on 2017-07-19 17:02:30
````
Well, at least for my example seems to be true\
x(1) = 0.6\*x(o) + e(1)\
x(2) = 0.6\*(0.6\*x(o) + e(1))+e(2)\
Since E\[e(1)\] = E\[e(2)\] = 0 we have E\[x(2)\] = 0.36\*x(0)\
Analogously, E\[x(3)\] = 0.216\*x(0) and so on \... so we quickly decay
to 0.

I will try to test what we get if there is an MA-term, then the decay
speed should be not so fast\...

**Reply by robjhyndman** [reply]
```header
Originally posted by tabita on 2017-07-26 15:29:39
````
Hai, i want to ask about forecast with arima subset.

Can i use r package for forecasting arima subset?

thank you

**Reply by robjhyndman** [reply]
```header
Originally posted on 2017-07-27 10:26:15
````
If you mean a subset of AR and MA terms, then yes, but you need to
specify which ones. See the fixed argument in the stats::arima function.

**Reply by robjhyndman** [reply]
```header
Originally posted by tabita on 2017-07-27 11:56:00
````
Hi, thank you for your reply. Do I need to choose between AR or MA? I
wonder if I can use them both. If it\'s not possible to use them both at
once, I would be really grateful if you could help me show the theory
(that I need to use AR and MA separately) and the code example to choose
which method is better. Thank you so much for your help.

**Comment by robjhyndman** [comment]
```header
Originally posted by Robert Crompton on 2017-07-22 20:25:11
````
Wow, this is funny - in the humorous sense, not the odd sense. There is
a Marxist quote (Groucho, not Karl) that I have adapted:

These are my models. If you don\'t like the results, I have others.

**Comment by robjhyndman** [comment]
```header
Originally posted by joshua hemmingway on 2018-09-06 18:59:09
````
I am having this issue, however upon a Box Cox transformation (Guerrero
method), the forecast drastically changes and all of a sudden an upward
trend and seasonality appears for future forecasts. However, time series
cross validation indicates that the auto.arima model selection on the
untransformed data yields the smallest MSE relative to cross validation
for the model with the transformed response \-- does this lower MSE
arise because the response variables are different? What should be done
for model selection under this scenario?

**Comment by amarjitsinghchandhial** [comment]
Hi Rob,

Please define an IID model.

Also, why would an AR model generate a straight-line forecast? I appreciate that dynamic multistep ahead forecasts will tend to the E(y(t)) i.e. long-term mean for a stationary AR model, but why as a straight-line?

Amarjit

**Reply by robjhyndman** [reply]
An iid model is where every observation is an independent draw from the same distribution. IID stands for "independent and identically distributed".
An AR model does not generate a straight-line forecast, and I didn't say otherwise in this post.
